yt_dlp/extractor/slideslive.py

```diff
@@ -127,8 +198,31 @@ def _extract_custom_m3u8_info(self, m3u8_data):
         return m3u8_dict
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        webpage = self._download_webpage(url, video_id)
+        video_id, original_query = self._match_id(url), parse_qs(url)
+        webpage, urlh = self._download_webpage_handle(
+            f'https://slideslive.com/embed/presentation/{video_id}', video_id,
+            headers=traverse_obj(original_query, {
+                'Referer': ('embed_parent_url', ...),
+                'Origin': ('embed_container_origin', ...),
+            }, get_all=False), query=traverse_obj(original_query, {
+                'embed_parent_url': 'embed_parent_url',
+                'embed_container_origin': 'embed_container_origin',
+            }))
```

Comment:
I would rather have the parameters be explicitly defined:
```py
    def _download_embed_webpage_handle(self, video_id, parent_url, container_origin):
        return self._download_webpage_handle(
            f'https://slideslive.com/embed/presentation/{video_id}', video_id,
            headers={'Referer': parent_url, 'Origin': container_origin},
            query={'embed_parent_url': parent_url, 'embed_container_origin': container_origin})
```
```py
        original_query = parse_qs(url)
        webpage, urlh = self._download_embed_webpage_handle(
            video_id, traverse_obj(original_query, ('embed_parent_url', 0)),
            traverse_obj(original_query, ('embed_container_origin', 0)))
```
```py
            webpage, _ = self._download_embed_webpage_handle(
                video_id, f'https://{domain}/', f'https://{domain}')
```

Reply:
We won't always have values for `parent_url` and `container_origin`, so I think it's easier to return a dict from `traverse_obj` and have it filter `None` for us. Otherwise this way would be better